Android/app/src/main/java/io/github/project_travel_mate/utilities/ChecklistFragment.java

```diff
@@ -99,14 +102,35 @@ void onClick() {
     public void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setHasOptionsMenu(true);
+
+    }
+
+    //get called every time whenever invalidateMenuoptions get called
+    @Override
+    public void onPrepareOptionsMenu(Menu menu) {
+        super.onPrepareOptionsMenu(menu);
+        hideShowMenuItem();
+        MenuItem item = menu.findItem(R.id.action_delete);
+
+        if (mItems.size() > 0)
+            item.setVisible(true);
+        else
+            item.setVisible(false);
+
+
     }
 
+    //get called only once
     @Override
     public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
         getActivity().getMenuInflater().inflate(R.menu.checklist_menu, menu);
+        getActivity().invalidateOptionsMenu();
```

Comment:
I am suggesting an alternate approach to fix this whole issue.

Just create a private variable `Menu mMenu` & assign it the value in this function.

Now, in the code, you can just hide or unhide the delete action using this menu only.

@kbisht84

Reply:
ok let me try that ..thanks @Swati4star